build-scripts/print-patches-for.py

```diff
@@ -72,22 +73,29 @@ def get_patches_for(component: str, version_string: str) -> list:
     component_version = Version(version_string)
     component_dir = DIR / "components" / component
 
-    patches_dir = component_dir / "patches"
-    if not patches_dir.is_dir():
-        return []
-
-    candidates = [
-        Version(path.name)
-        for path in sorted(
-            patches_dir.iterdir(), reverse=True
-        )  # sort by reverse to handle exact matches
-        if path.is_dir()
-    ]
-    patch_version = find_suitable_patch_version(candidates, component_version)
-    if patch_version is None:
-        return []
-
-    return [p.resolve().as_posix() for p in (patches_dir / patch_version).iterdir()]
+    patches = []
```

Comment:
I probably miss something here but this looks like all patches (`patches` and `strict-patches`) are loaded and applied. Where do we check if the snap is strict and only then apply the strict patches?

Reply:
For classic builds, we'll change https://github.com/canonical/k8s-snap/pull/95/files#diff-e5250e12ec707f3ef82f07821039437b2e4fc1aa76e04504069abd4baae255d7R8 to only check in the `patches` folder instead.

We could parse the snapcraft.yaml, but that breaks remote builds (since snapcraft.yaml is out of the source dir of the part)